src/transaction.ts

```diff
@@ -462,13 +462,15 @@ export class Snapshot extends EventEmitter {
         ) => {
           if (err) {
             setSpanError(span, err);
-            span.end();
-            callback!(err, resp);
-            return;
+          } else {
+            this._update(resp);
           }
-          this._update(resp);
+
+          // begin simply issues an RPC to signal a state change to the backend,
```

Comment:
I think its clear that we need `span.end()` here, can we remove this comment.

Reply:
Sure, but the reason that I added that comment is to help the next maintainer know why in some places we shall end the span only after the transaction is ended/errored vs why in some we end the span before returning.